app/koi-client/src/hook/query/Stock/useRecommendedPartners.ts

```diff
@@ -0,0 +1,47 @@
+import { useAtomValue } from 'jotai';
+import { objectEntries } from '@toss/utils';
+import { useMemo } from 'react';
+import { Query } from '../..';
+import { UserStore } from '../../../store';
+
+export const useRecommendedPartners = (stockId: string | undefined): { partnerNicknames: string[] } => {
+  const supabaseSession = useAtomValue(UserStore.supabaseSession);
+  const userId = supabaseSession?.user.id;
+
+  const { data: stock } = Query.Stock.useQueryStock(stockId);
+  const { data: users } = Query.Stock.useUserList(stockId);
+  const { data: profiles } = Query.Supabase.useQueryProfileById(users?.map((v) => v.userId) ?? []);
+
+  const partnerNicknames = useMemo(() => {
+    if (!stock || !userId || !profiles?.data) return [];
+
+    const [partnerIds] = objectEntries(stock.companies).reduce(
+      (reducer, [_, companyInfos]) => {
+        const [partnerIds] = reducer;
+
+        companyInfos.forEach((_, idx) => {
+          if (companyInfos[idx].정보.some((name) => name === userId)) {
+            const partner = companyInfos[idx].정보.find((name) => name !== userId);
+            if (partner && !partnerIds.some((v) => v === partner)) {
+              partnerIds.push(partner);
+            }
```

Comment:
```suggestion
            const partners = companyInfos[idx].정보.filter((name) => name !== userId);
            partners.forEach((partner) => {
              if (partner && !partnerIds.some((v) => v === partner)) {
                partnerIds.push(partner);
              }
            });
```

파트너의 정의는 "나랑 같은 정보를 가진 사람들"인데,
랜덤 뽑기 기능의 추가로 인해, 언제든지 파트너가 추가될 수 있어서
위와 같이 제안을 드려요!

Reply:
내일 라이브라서 바로 suggestion 적용할게요!